### PR TITLE
[CDAP-17246] Fix js error while editing draft or importing pipeline from wrangler

### PR DESCRIPTION
**Pull request by ajainarayanan**
JIRA - https://issues.cask.co/browse/CDAP-17246

**Cause**
- This PR https://github.com/cdapio/cdap/pull/12762 fixes importing pipeline from 6.1.x version of CDAP to 6.2. 
- However this change introduced issues with importing pipeline from wrangler or editing drafts
- We introduced 'id' as part of every node in the pipeline being imported. We didn't however fix the path where users could import pipeline from wrangler or edit drafts.

**Testing**
- This PR fixes all possible places where an individual node name/label would be used or modified.
  - Import via UI
  - Import/Edit via wrangler
  - Edit drafts
  - Copy/Paste nodes (editing pipeline)
  - Add/Delete nodes
  - Group delete nodes
  - Undo/Redo actions (deleting and undoing delete)
  - Duplicate from pipeline details
  - Preview (run and fetch data for nodes with spaces in their labels)

**Side-Effect**
- https://github.com/cdapio/cdap/pull/12749 This PR fixes fetching preview for individual stages by changing label to name. Since we need to use labels (user identifiable names for nodes) I have restored the original behavior of sending in labels for preview. We need to make sure the node labels are properly used in constructing the pipeline.

### CODE REVIEW

**Comment by ajainarayanan** [comment]
I give up on e2e test plan. Ran tests locally and fixed all flaky tests. We have all tests passing.

Screen Shot 2020-09-14 at 4 20 38 PM
Screen Shot 2020-09-14 at 3 40 35 PM

File: cdap-ui/app/hydrator/services/hydrator-service.js [hunk 1]
```diff
@@ -41,7 +41,8 @@ class HydratorPlusPlusHydratorService {
       let nodeInfo = angular.extend(stage, {
         type: stage.plugin.type,
         label: stage.plugin.label,
-        icon: this.DAGPlusPlusFactory.getIcon(stage.plugin.name)
+        icon: this.DAGPlusPlusFactory.getIcon(stage.plugin.name),
+        id: stage.id || `${stage.name.replace(/[ \/]/g, '-')}${Math.floor(Math.random() * 100)}`,
```
Comment (elfenheart) [inline cdap-ui/app/hydrator/services/hydrator-service.js]: why not use uuid instead of doing random?

**Review by elfenheart — APPROVED** [review]
👍  LGTM

**Comment by yukiej** [comment]
lgtm!

**Review by njbriggs — APPROVED** [review]
Code LGTM. I'll try my use case again in the bug bash.

**Pull request merged**